content/blog/2022-08-29-announcing-poetry-1-2-0.md

```diff
@@ -0,0 +1,636 @@
+{{% /note %}}
+
+#### Optional groups
+
+A dependency group can be declared as optional. This makes sense when you have a group of dependencies that are only
+required in a particular environment or for a specific purpose.
+
+```toml
+[tool.poetry.group.docs]
+optional = true
+
+[tool.poetry.group.docs.dependencies]
+mkdocs = "*"
+```
+
+Optional groups can be installed in addition to the **default** dependencies by using the `--with` option of
+the [`install`]({{< relref "docs/cli#install" >}}) command.
+
+```bash
+poetry install --with docs
+```
+
+{{% warning %}}
+Optional group dependencies will **still** be resolved alongside other dependencies, so special care should be taken to
+ensure they are compatible with each other.
+{{% /warning %}}
+
+#### Adding a dependency to a group
+
+The [`add`]({{< relref "docs/cli#add" >}}) command is the preferred way to add dependencies to a group. This is done by
+using the `--group (-G)` option.
+
+```bash
+poetry add pytest --group test
+```
+
+If the group does not already exist, it will be created automatically.
+
+#### Installing group dependencies
+
+**By default**, dependencies across **all groups** will be installed when executing `poetry install`.
+
+You can **exclude** one or more groups with the `--without` option:
+
+```bash
+poetry install --without docs,test
+```
+
+You can also opt in [optional groups]({{< relref "#optional-groups" >}}) by using the `--with` option:
+
+```bash
+poetry install --with docs
+```
+
+If you only want to install the **default**, non-grouped, dependencies, you can do so with the `--only` option:
+
+```bash
+poetry install --only main
+```
+
+Finally, if you wish to install **only specific groups** of dependencies without installing the default dependencies,
+the `--only` option can be used like so:
+
+```bash
+poetry install --only docs,test
+```
+
+#### Removing dependencies from a group
+
+The [`remove`]({{< relref "docs/cli#remove" >}}) command supports a `--group` option to remove packages from a specific
+group:
+
+```bash
+poetry remove mkdocs --group docs
+```
+
+### New plugin system
+
+Poetry now supports a plugin system to alter or expand Poetry's functionality.
+
+For example if your environment poses special requirements on the behaviour of Poetry which do not apply to the majority
+of its users or if you wish to accomplish something with Poetry in a way that is not desired by most users.
+
+In these cases you could consider creating a plugin to handle your specific logic.
+
+#### Using plugins
+
+Installed plugin packages are automatically loaded when Poetry starts up.
+
+There are multiple ways to install plugins for Poetry, the easiest one (that should account for all the ways Poetry can
+be installed) being:
+
+```bash
+poetry self add poetry-<PLUGIN_NAME>
+```
+
+The `plugin add` command will ensure that the plugin is compatible with the current version of Poetry and install the
```

Comment:
```suggestion
The `self add` command will ensure that the plugin is compatible with the current version of Poetry and install the
```

Reply:
Woops, that's what happens when you only grep for `poetry plugin` :smile: 
Thanks!